### PR TITLE
enable nullable for new projects

### LINKED ISSUE #485: Enable nullable by default for new projects

### PR DESCRIPTION
**Pull request by jmarolf**
resolves https://github.com/dotnet/templates/issues/485

### CODE REVIEW

File: src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Pages/Error.cshtml [hunk 1]
```diff
@@ -18,7 +18,7 @@
             <h1 class="text-danger">Error.</h1>
             <h2 class="text-danger">An error occurred while processing your request.</h2>
 
-            @if (Model.ShowRequestId)
+            @if (Model?.ShowRequestId == true)
```
Comment (JamesNK) [inline src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Pages/Error.cshtml:21]: nit: Personally I do this for null in boolean checks:
```suggestion
            @if (Model?.ShowRequestId ?? false)
```

File: src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Pages/Error.cshtml [hunk 2]
```diff
@@ -18,7 +18,7 @@
             <h1 class="text-danger">Error.</h1>
             <h2 class="text-danger">An error occurred while processing your request.</h2>
 
-            @if (Model.ShowRequestId)
+            @if (Model?.ShowRequestId == true)
```
Comment (JamesNK) [inline src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Pages/Error.cshtml:21]: See the previous comment for all repeats.

**Comment by mkArtakMSFT** [comment]
@danroth27 can you please confirm that we want to take this change in for our templates? Thanks!

**Comment by pranavkm** [comment]
I would love to enable nullability, but I think we would enable it by default rather than make it a switch that a user has to choose. That said, I think there are some important pieces we would need to address before we did that:

* Nullability for identity UI so that it scaffolds well
* Update scaffolding for controllers / ef /razor
* Fix some Razor code generation bugs where it inadvertently suppresses nullability in user code.

The remainder of the framework is annotated enough (and we will cover more by 6.0) that users should find the experience productive.

**Comment by jmarolf** [comment]
@pranavkm should this be closed or is there some set of projects we think its appropriate for?

**Comment by danroth27** [comment]
> I would love to enable nullability, but I think we would enable it by default rather than make it a switch that a user has to choose.

:+1:

> I think there are some important pieces we would need to address before we did that:

Do we have tracking issues for these remaining work items?

**Comment by jmarolf** [comment]
> I would love to enable nullability, but I think we would enable it by default rather than make it a switch that a user has to choose.

it is enabled by default. The `defaultValue` is `true`. The user needs to explicitly pass false

**Comment by captainsafia** [comment]
@danroth27 There's https://github.com/dotnet/aspnetcore/issues/32783

**Comment by danroth27** [comment]
@jmarolf I think the question is do we need a switch at all. Once we are at a place with ASP.NET Core that we can turn on nullability checks by default, is there really a need for users to have a switch to turn it off?

**Comment by jmarolf** [comment]
> I think the question is do we need a switch at all

I have no objections. I added the switch as a conservative measure. If we are comfortable just using this as the default that makes things simpler.

File: src/ProjectTemplates/Web.Spa.ProjectTemplates/content/React-CSharp/Pages/Error.cshtml [hunk 3]
```diff
@@ -7,7 +7,7 @@
 <h1 class="text-danger">Error.</h1>
 <h2 class="text-danger">An error occurred while processing your request.</h2>
 
-@if (Model.ShowRequestId)
+@if (Model?.ShowRequestId ?? false)
```
Comment (davidfowl) [inline src/ProjectTemplates/Web.Spa.ProjectTemplates/content/React-CSharp/Pages/Error.cshtml:10]: this looks kinda terrible.

**Comment by DamianEdwards** [comment]
Nullable was enabled by the other changes that went in today.

**Pull request closed without merging**